### PR TITLE
Pushing test for Persistent XSS in HTML

### PR DESCRIPTION
**Pull request by SeheX**

### CODE REVIEW

File: src/test/java/org/sasanlabs/service/vulnerability/xss/reflected/PersistentXSSInHTMLTagVulnerabilityTest.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        ResponseEntity<String> response = vulnerability.getVulnerablePayloadLevel6(queryParams);
+
+        verify(postRepository, times(1)).save(any());
+
+        assertEquals(200, response.getStatusCodeValue());
+    }
+
+    @Test
+    public void testGetVulnerablePayloadLevel7() {
+        Map<String, String> queryParams = new HashMap<>();
```
Comment (preetkaran20) [inline src/test/java/org/sasanlabs/service/vulnerability/xss/reflected/PersistentXSSInHTMLTagVulnerabilityTest.java:117]: i think we can use collections.singletonMap

File: src/test/java/org/sasanlabs/service/vulnerability/xss/reflected/PersistentXSSInHTMLTagVulnerabilityTest.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        vulnerability = new PersistentXSSInHTMLTagVulnerability(postRepository);
+    }
+
+    @Test
+    public void testGetVulnerablePayloadLevel1() {
+        Map<String, String> queryParams = Collections.singletonMap("comment", "<script>alert('XSS')</script>");
+
+        ResponseEntity<String> response = vulnerability.getVulnerablePayloadLevel1(queryParams);
+
+        verify(postRepository, times(1)).save(any());
```
Comment (preetkaran20) [inline src/test/java/org/sasanlabs/service/vulnerability/xss/reflected/PersistentXSSInHTMLTagVulnerabilityTest.java:37]: shall we also assert on the arguments passed to the postRepository?

File: src/test/java/org/sasanlabs/service/vulnerability/xss/reflected/PersistentXSSInHTMLTagVulnerabilityTest.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+        ResponseEntity<String> response = vulnerability.getVulnerablePayloadLevel5(queryParams);
+
+        verify(postRepository, times(1)).save(any());
+
+        assertEquals(200, response.getStatusCodeValue());
+    }
+
+    @Test
+    public void testGetVulnerablePayloadLevel6() {
```
Comment (preetkaran20) [inline src/test/java/org/sasanlabs/service/vulnerability/xss/reflected/PersistentXSSInHTMLTagVulnerabilityTest.java:98]: I think adding a few tests would help here like input having NullByte and its impact on level 6 and level 5.

File: src/test/java/org/sasanlabs/service/vulnerability/xss/reflected/PersistentXSSInHTMLTagVulnerabilityTest.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        assertEquals(200, response.getStatusCodeValue());
+    }
+
+    @Test
+    public void testGetVulnerablePayloadLevel4() {
+        Map<String, String> queryParams = Collections.singletonMap("comment", "<img src='x' onerror='alert(1)'>");
+
+        ResponseEntity<String> response = vulnerability.getVulnerablePayloadLevel4(queryParams);
+
+        verify(postRepository, times(1)).save(any());
```
Comment (preetkaran20) [inline src/test/java/org/sasanlabs/service/vulnerability/xss/reflected/PersistentXSSInHTMLTagVulnerabilityTest.java:81]: I think we need to add a few more granular level of assertions then just verifying the call to postRepository.

**Review by preetkaran20 — CHANGES_REQUESTED** [review]

**Comment by SeheX** [comment]
Can you check ? Is it good like this ?

**Review by preetkaran20 — APPROVED** [review]
LGTM. Thanks a lot for the PR.

**Comment by preetkaran20** [comment]
@SeheX Thanks a lot for the PR !!!

**Pull request merged**